syntax = 'proto3';

package auth;

service AuthService {
  rpc GetPayload(GetPayloadRequest) returns (GetPayloadResponse);
  rpc GetUser(GetUserRequest) returns (GetUserResponse);
}

message GetPayloadRequest {
  string token = 1;
}

message GetPayloadResponse {
  string user_id = 1;
  string role = 2;
  string expires_at = 3; // ISO 8601 format
  string service_name = 4;
  repeated string permissions = 5;
}

message GetUserRequest {
  string token = 1;
}

message GetUserResponse {
  string id = 1;
  string external_id = 2;
  string role = 3;
  string external_role = 4;
  string name = 5;
  string surname = 6;
  string patronymic = 7;
  string email = 8;
  string faculty = 9;
  string login = 10;
  string last_login = 11;
  string created_at = 12;
}
